### ISSUE TITLE
Add "content" category in commit message guidelines

### ISSUE THREAD

**Issue by pensierinmusica** [angular/angular.js#14210]
Hi,

I've been using the Angular commit messages convention fro more than a year now, and I find it great.

On the other hand, I think there's one type of commits that is very relevant, and it's missing from the list: the "content" type.

Every time I'm working on an app or a website and I need to commit changes to its static content (e.g. text in a page, pictures, titles, etc.) I can't find a good fit in the existing commit types included in the list.

In fact, it's neither "feat", nor "style", nor "docs". Apart from this, I've found the list to be pretty exhaustive.

Therefore, I propose to include the "content" type in the list, with the description "A content change that only affects static text or images" (or something similar).

Let me know your thoughts, and if you'd like me to create a PR for it!

Thanks, cheers :)

p.s. relates to stevemao/conventional-changelog-angular#3

**Comment by Narretz** [comment]
I think the commit message guidelines don't apply very well to "app" projects. Angular is a framework, so there's not going to be a lot of content. Although we usually use "chore" for content changes on angularjs.org. But I think a content category is out of scope for the commit message guidelines.

**Comment by gkalpak** [comment]
@Narretz is right, althought it might make sense for a website, we don't have any "content" in this repo. The only content we have is docs (so the `docs` type is better suited).

Maybe you were thinking about [angularjs.org](https://github.com/angular/angularjs.org) (which is separate repo) ?
(But even there, there's no "content" in the tranditional sense of the term, imo.)

**Comment by pensierinmusica** [comment]
What about the website at http://angular.io ?

**Comment by pensierinmusica** [comment]
Btw, I understand Angular is a framework and as such has no "content". On the other hand, it looks like @stevemao won't change [conventional-changelog-angular](https://github.com/stevemao/conventional-changelog-angular/issues/3) unless you approve such a category, which in turn means it can't be integreated as default in [cz-conventional-changelog](https://github.com/commitizen/cz-conventional-changelog/issues/15).

Imho the problem is that following the angular convention was a great start, but we can't block all projects in the world from having a useful "content" category because Angular is a framework.

Therefore there are really two solutions I can see to the problem:

1. The Angular team allows the "content" category to be integrated in `conventional-changelog-angular`.
2. `cz-conventional-changelog` detaches itself from pure Angular standards

@gkalpak and @Narretz thoughts?

**Comment by gkalpak** [comment]
I'm not sure aboutt their commit message conventions. It's a different repo as well.
Why don't you post an issue there ? Maybe it's a good fit for them.

BTW, here are the main repos:

* `angular.js` (ng1): https://github.com/angular/angular.js
* `angularjs.org` (ng1 website): https://github.com/angular/angularjs.org
* `angular` (ng2): https://github.com/angular/angular
* `angular.io` (ng(2) website): https://github.com/angular/angular.io

**Comment by gkalpak** [comment]
**EDIT:** This comment it a reply to https://github.com/angular/angular.js/issues/14210#issuecomment-195013441.

I don't think either ng1 or ng2 use [conventional-changelog-angular][1], but there have been discussions about switching to it (IIRC).

The Angular Team does not decide what goes into those projects; only their authors do.
I agree that what makes sense for Angular is not making sense for all projects in the world and they shouldn't be blocked (but the Angular Team is definitely not blocking anyone).

Nor does it make sense for Angular to incorporate any commit message type that might be useful for any project in the world.

--
IMO, (if that's not alredy the case) there should be several "presets" available, e.g. one for applications, another one for libraries, a third one for websites and so on. Different commit message types make sense for different types of projects.

[1]: https://github.com/stevemao/conventional-changelog-angular

**Comment by pensierinmusica** [comment]
@jimthedev I just thought that since `cz-conventional-changelog` is not only used by the Angular team, it makes sense to have the "content" category available **as default**.

Would you pease consider it? So that people don't need custom configurations to have it.

Then, I know it's possible to create a custom adapter, and thanks for being available to include it in the list. If there's no other way, I'll end up doing this. On the other hand, I'd still invite everyone to reconsider having the "content" category available in `cz-conventional-changelog` by default as the best course of action. Thoughts? Cheers!

**Comment by jimthedev** [comment]
@pensierinmusica I appreciate you asking so nicely. I have a hard time saying no to ideas so hopefully you know how tough this is.  I certainly understand why it seems like this would be a good idea. The thing is, my goal has always been to keep cz-conventional-changelog as close to the angular specs as possible since that is how it started and people rely on that assumption.  I should have named it better, but the key assumptions remain in place for the 25k users that we have per month of that adapter.

It would take a breaking change of that adapter to accomplish this change and as such I don't think that adapter will ever diverge. Partially this is to make sure integration with cz-customizable and semantic-release are easy and consistent, partially it is because this is not a default adapter but rather it is a reference implementation. A reference implementation is meant to show people how they could structure their own adapters. You do not need to use cz-conventional-changelog with commitizen, though many people do. Instead, please fork the reference implementation and make the changes you have for your specific use cases. Chances are that there are others wanting this change, but I am unwilling to make a breaking changes when we don't have an idea of how many people would even use such a feature. If it turns out that your implementation is the preferred then we can change the recommendation in the docs to use that adapter.

TLDR; There is no default commitizen adapter. Adapters are chosen by repo authors. Many use a reference implementation (cz-conventional-changelog), many do not.

**Comment by stevemao** [comment]
@pensierinmusica if you look at the issues at conventional-changelog, people always have their "own angular convention" which is slightly different than the real "angular convention". And most of the rules they make up are different than each other. I think the real "angular convention" works for most repos and it's very easy to modify on top of the angular preset (No need to fork, just `require` it and modify).

**Comment by stevemao** [comment]
However, if you ask if there is an easier way to config on top of another config (preset/adapter) the tool authors might think about that. EG: forking a repo might end up many duplicated code and force to sync with upstream etc, I think `require` would be better (`npm` is an awesome package manager) (I'm not pointing at commitizen since I know little about it. I'm trying to give an example and talking about similar tools in general). I've always trying to achieve this in my projects.

**Comment by pensierinmusica** [comment]
@jimthedev thanks for the explanation! I see, I thought that `cz-conventional-changelog` was the default adapter in case there was no adapter specified. Since it turns out that the user always needs to specify one, now I understand you rpoint much better. I guess we can close this issue, and the related ones, since we got to a fairly stable point. Thanks everyone for participating and helping out!

**Comment by jimthedev** [comment]
@pensierinmusica Great! Glad to have the discussion! :+1:
